File: caesar.c
#include<stdio.h>
#include<cs50.h>
#include<stdlib.h>
#include<ctype.h>

int main(int argc, string argv[])
{
    if(argc!=2)
    {
        printf("No comand line input \n");
        return 1;
    }
    int k;
    k=atoi(argv[1]);
    k=k%26;
        printf("plaintext: ");
        string pt=get_string();
        printf("ciphertext: ");
        for(int i=0;pt[i]!='\0';i++)
        {
            if(isalpha(pt[i]))
            {
            if((pt[i]+k)>'z')
            {
                printf("%c",(((pt[i]+k)%122)+96));
            }
            else if((pt[i]+k)>'Z' && (pt[i]<='Z'))
            {
                printf("%c",((pt[i]+k)%90)+64);
            }
            else{
             printf("%c",pt[i]+k);   
            }}
            else
            {
                printf("%c",pt[i]);
            }
        }
        printf("\n");
}

File: caesar.py
from cs50 import *
from sys import *

if len(argv)!=2:
    print("No comand line input")
    exit(1)
k=int(argv[1])
k=k%26
otp=[]
inp=get_string("plaintext: ")
for i in range(len(inp)):
    ind=ord(inp[i])
    if ind>=97 and ind<=122:
        if ind+k>122:
            nid=(ind+k)%123+97
        else:
            nid=(ind+k)
        otp.append(chr(nid))
    elif ind>=65 and ind <=90:
        if ind+k>90:
            nid=(ind+k)%91+65
        else:
            nid=ind+k
        otp.append(chr(nid))
    else:
        otp.append(inp[i])
print("ciphertext: "+''.join(otp))
